akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -613,6 +613,9 @@ final case class RunnableGraph[+Mat](override val traversalBuilder: TraversalBui
    */
   override def async(dispatcher: String, inputBufferSize: Int): RunnableGraph[Mat] =
     super.async(dispatcher, inputBufferSize).asInstanceOf[RunnableGraph[Mat]]
+
+  /** Converts this Scala DSL element to it's Java DSL counterpart. */
+  def asJava[JMat >: Mat]: javadsl.RunnableGraph[JMat] = javadsl.RunnableGraph.fromGraph(this)
```

Comment:
Test coverage for this as well?

Reply:
Yup. Two of the added test-cases call this.

Reply:
Does this `JMat >: Mat` work as intended.

I guess the purpose is to be able to do this:
```
final akka.stream.scaladsl.RunnableGraph<String> scalaRunnable = null;
RunnableGraph<Object> objectRunnableGraph = scalaRunnable.asJava();
```

and that compiles, but the following also compiles:
```
final akka.stream.scaladsl.RunnableGraph<Object> scalaRunnable = null;
RunnableGraph<String> objectRunnableGraph = scalaRunnable.asJava();
```

I see that we have such in other `asJava` as well. All of the following compiles and that can't be correct?
```
    akka.stream.scaladsl.Source<String, NotUsed> scaladslSource1 = null;
    Source<String, NotUsed> source1 = scaladslSource1.asJava();
    Source<Object, NotUsed> source2 = scaladslSource1.asJava();
    Source<Integer, NotUsed> source3 = scaladslSource1.asJava();

    akka.stream.scaladsl.Source<Object, NotUsed> scaladslSource4 = null;
    Source<String, NotUsed> source4 = scaladslSource1.asJava();
```

Reply:
I might have added too many test-cases. :) I should not have added the scaladsl tests from Java, because scaladsl methods use lower type bounds and that does not work from Java. This is the continuation of https://github.com/akka/akka/issues/24368

Reply:
Good catch Patrik, this is the same as we had before, although I think this one is perhaps less scary than the other one since it is so seldom used.